pkg/controller/bypass/bypass_controller.go

```diff
@@ -0,0 +1,277 @@
+package bypass
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	nd "istio.io/istio/cni/pkg/nodeagent"
+	"istio.io/istio/pkg/util/sets"
+	"kmesh.net/kmesh/pkg/logger"
+	"kmesh.net/kmesh/pkg/utils"
+)
+
+var (
+	log = logger.NewLoggerField("bypass")
+	FS  embed.FS
+)
+
+func getCurrentNodeName(client kubernetes.Interface, podName, namespace string) (string, error) {
+	pod, err := client.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error getting pod %s in namespace %s: %v", podName, namespace, err)
+	}
+
+	nodeName := pod.Spec.NodeName
+
+	return nodeName, nil
+}
+
+func StartPodEventWatcher(client kubernetes.Interface, podName, podNamespace string) error {
+	stopChan := make(chan struct{})
+
+	nodeName, err := getCurrentNodeName(client, podName, podNamespace)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	pod, err := client.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	log.Infof("pod %s, pod ns %s", pod.GetNamespace(), pod.GetName())
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, 30*time.Second,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+			options.LabelSelector = "kmesh.net/bypass=enabled"
```

Comment:
If we add this as a label selector, we cannot handle bypass enabled to disabled case